community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -129,10 +130,49 @@ public RotatingFileOutputStreamSupplier( FileSystemAbstraction fileSystem, File
         this.rotationListener = rotationListener;
         this.rotationExecutor = rotationExecutor;
         this.outRef.set( openOutputFile() );
+        // Wrap the actual reference to prevent race conditions during log rotation
+        this.streamWrapper = new OutputStream()
+        {
+            @Override
+            public void write( int i ) throws IOException
+            {
+                synchronized ( outRef )
```

Comment:
Now we always synchronize on `outRef`. Do we still need it to be an `AtomicReference`?